src/jdk.compiler/share/classes/com/sun/tools/javac/code/Type.java

```diff
@@ -396,7 +396,7 @@ public <M extends TypeMetadata> M getMetadata(Class<M> metadataClass) {
     @SuppressWarnings("unchecked")
     public <M extends TypeMetadata, Z> Z getMetadata(Class<M> metadataClass, Function<M, Z> metadataFunc, Z defaultValue) {
         for (TypeMetadata m : metadata) {
-            if (m.getClass() == metadataClass) {
+            if (m.getClass().isAssignableFrom(metadataClass)) {
```

Comment:
isn't TypeMetadata `sealed` ? I suppose the code you wrote is correct (and better than `==`), but I don't think we should have inherited metadata? E.g. an `equals` should be good?

Reply:
yes I guess it should be good, I will replace it in the next iteration